lib/AST/DistributedDecl.cpp

```diff
@@ -318,7 +318,7 @@ swift::getDistributedSerializationRequirements(
 
   if (auto alias = dyn_cast<TypeAliasType>(existentialRequirementTy.getPointer())) {
     auto ty = alias->getDesugaredType();
-    if (isa<ClassType>(ty) || isa<StructType>(ty) || isa<EnumType>(ty)) {
+    if (ty && !ty->isExistentialType()) {
```

Comment:
I don't think you need to do `dyn_cast<TypeAliasType>(...)` since `existentialRequirementTy->isExistentialType()` is going to desugar typealias.

Reply:
I think I’ve seen nullpointer errors then, will check!

Reply:
Seems it works, thanks for calling it out :-)